### PR TITLE
circulation: hide buttons if action isn't available.

### PR DESCRIPTION
**Pull request by zannkukai**
If the patron can't operate an action due to circulation restrictions
(patron blocked, patron type limits, ...), circulation buttons should be
hide. In the patron profile view, adds warning/error messages depending
of patron restrictions.

Closes rero/rero-ils#1357
Closes rero/rero-ils#1356

Co-Authored-by: Renaud Michotte <renaud.michotte@gmail.com>

## How to test?

image

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?

### CODE REVIEW

**Review by Garfield-fr — APPROVED** [review]

File: rero_ils/modules/patrons/templates/rero_ils/_patron_profile_requests.html [hunk 1]
```diff
@@ -49,7 +49,7 @@
         <div class="col-lg-2{% if loan.state == 'ITEM_AT_DESK' %} text-success{% endif %}">
           {% if loan.state == 'ITEM_AT_DESK' %}
           <i class="fa fa-check"  title="{{ _('item at desk') }}" aria-hidden="true"></i>
-          {{ _('ready') }}
+          {{ _('to pick uup') }}
```
Comment (lauren-d) [inline rero_ils/modules/patrons/templates/rero_ils/_patron_profile_requests.html:52]: change `to pick uup` to `to pick up` 🙃?

**Review by lauren-d — CHANGES_REQUESTED** [review]

**Comment by ManaDeweerdt** [comment]
When I try to checkout an item I get an error 'item not found'.
But I can checkin and checkout an item already borrowed.

**Comment by ManaDeweerdt** [comment]
Like the https://github.com/rero/rero-ils/issues/1357 the button 'request' should be visible even he has the limits reached, followed by an explanation.

**Comment by benerken** [comment]
Great job Renaud ! 
some comments for now:

- If we define a general checkout limit only => the system displays a message when you arrive on the patron account in the professional view  AND  It displays a flash message when you do a checkout  => OK
(A permanent message is displayed in the patron public view) => OK

- If we define a general checkout limit AND a library checkout limit => the system doesn't display a permanent message when you arrive on the patron account in the professional view. It displays a flash message when you do a checkout. Why ? 
(Same behaviour in the patron public view => No permanent message)

**Review by iGormilhit — CHANGES_REQUESTED** [review]
Commit message body: replace twice *disable* by *disabled*. Thanks.

**Comment by zannkukai** [comment]
> Like the #1357 the button 'request' should be visible even he has the limits reached, followed by an explanation. [enhancement]

It's more difficult :: One of the main reason why an item is not requestable is because the item is not in the same organisation than patron. In this case, it's not user friendly to get the button disabled with this message.

**Review by lauren-d — APPROVED** [review]

**Comment by pronguen** [comment]
* Wording
  * The unit of the field for each limit should be clearer in the field label (same problem than in cipo, https://github.com/rero/rero-ils/issues/1305).
  * I don't understand the word *default* in some of the labels.
  * The label should correspond to the description given below and does not need to repeat the info given in the toggle.
  * Suggestions:
    * Global checkout limit > Number of checkouts in the organisation
    * Default checkout limit by library > Number of checkouts by library
    * Default fee amount limit > Fee amount
    * Default overdue items limit > Number of overdue items
image
* Message in public interface (for patrons)
  * Don't display any message above, in the patron account (simplification), as they are note user-friendlyI think it is enough to display the message on the disabled buttons as tooltip.
    * Potential problem: the user does not know that he cannot request document. But if we display the request button in any case (sometimes disabled) it will solve this. See also Mana's comment [can be done later, issue can be opened].
  * Tooltips on buttons (request/renew): adapt if possible to end user, as specified in the US: "Renewal impossible: you have too many overdue items." / "Renewal impossible: you have too many fees." / etc.
  * Tootips in Google Chrome (Version 86.0.4240.183 (Build officiel) (64 bits)) is problematic: no tooltip for the first item to renew, other tooltip permanent (see screenshot)
  * This works on Firefox
image
* Circulation interface
  * Simplification of the flash message: the first part could be used as first line. In example below, use "Checkout denied" as first line.
image
  * [enhancement] Ideally, the messages display above in user checkout account are adapted. Now they are answer messages after an action tried.
* General
  * Limit by fee amount includes only overdue fees? Or it just does not include subscription fees? To decide, we may need to adapt the labels and messages.

**Pull request merged**